// handlebars-brunch/lib/index.js
(function() {
  var HandlebarsCompiler, handlebars, sysPath;

  handlebars = require('handlebars');

  sysPath = require('path');

  module.exports = HandlebarsCompiler = (function() {

    HandlebarsCompiler.prototype.brunchPlugin = true;

    HandlebarsCompiler.prototype.type = 'template';

    HandlebarsCompiler.prototype.extension = 'hbs';

    HandlebarsCompiler.prototype.pattern = /\.(?:hbs|handlebars)$/;

    function HandlebarsCompiler(config) {
      this.config = config;
      null;
    }

    HandlebarsCompiler.prototype.compile = function(data, path, callback) {
      var content, error, result;
      try {
        content = handlebars.precompile(data);
        return result = "module.exports = Handlebars.template(" + content + ");";
      } catch (err) {
        return error = err;
      } finally {
        callback(error, result);
      }
    };

    HandlebarsCompiler.prototype.include = [sysPath.join(__dirname, '..', 'vendor', 'handlebars.runtime-1.0.0.beta.6.js')];

    return HandlebarsCompiler;

  })();

}).call(this);
